manticore/core/state.py

```diff
@@ -364,3 +364,12 @@ def mem(self):
 
     def _init_context(self):
         self.context['branches'] = dict()
+
+    def generate_testcase(self, name, message='State generated testcase'):
+        """
+        Generate a testcase for this state and place in the analysis workspace.
+
+        :param str name: Short string identifying this testcase used to prefix workspace entries.
+        :param str message: Longer description
+        """
+        self.publish('state_generate_inputs', name, message)
```

Comment:
I just realized it really doesn't make any sense to create an entirely new `state_generate_inputs` event, and then connect it to an existing handler `_generate_testcase_callback`. I should be able to use the existing `will_generate_inputs` here